zio-http-gen/src/test/resources/ComponentAliasOrderId.scala

```diff
@@ -0,0 +1,9 @@
+package test.component
+
+import zio.prelude.Newtype
+import zio.schema.Schema
+import java.util.UUID
+
+object OrderId extends Newtype[UUID] {
+  implicit val schema: Schema[OrderId.Type] = Schema.primitive[UUID].transform(wrap, unwrap)
```

Comment:
You don't need to write them. They can be automatically derived:
```suggestion
  implicit val schema: Schema[Type] = derive
```

Reply:
This is also a part of #3002, not really changed in this PR.
I did tried it at first though:
https://github.com/zio/zio-http/pull/3002/commits/f11b05d5d71c03fb0675ae8447690d2d0aff0aa3#diff-05e59888caf2ce39f045befc5e4ffc68051f5a68cb3d001410545aa7ae2b96cbR7

Couldn't get it to work on all cross versions, so I went for the more verbose code.

I figured its generated code anyway, and not using the macro gives better compilation times, so there's a silver lining...

Perhaps in a future PR this can be improved, but as said - this is not a part of this PR.